drivers/kw2xrf/kw2xrf_netdev.c

```diff
@@ -689,24 +692,26 @@ static void _isr_event_seq_tr(netdev_t *netdev, uint8_t *dregs)
     }
 
     if (dregs[MKW2XDM_IRQSTS1] & MKW2XDM_IRQSTS1_SEQIRQ) {
+        DEBUG("[kw2xrf] SEQIRQ\n");
+        bool channel_was_busy = false;
         if (dregs[MKW2XDM_IRQSTS1] & MKW2XDM_IRQSTS1_CCAIRQ) {
             irqsts1 |= MKW2XDM_IRQSTS1_CCAIRQ;
             if (dregs[MKW2XDM_IRQSTS2] & MKW2XDM_IRQSTS2_CCA) {
                 DEBUG("[kw2xrf] CCA CH busy\n");
                 netdev->event_callback(netdev, NETDEV_EVENT_TX_MEDIUM_BUSY);
+                channel_was_busy = true;
             }
         }
 
         irqsts1 |= MKW2XDM_IRQSTS1_SEQIRQ;
         assert(dev->pending_tx != 0);
         dev->pending_tx--;
-
         if (dregs[MKW2XDM_IRQSTS3] & MKW2XDM_IRQSTS3_TMR3IRQ) {
             /* if the sequence was aborted by timer 3, ACK timed out */
             DEBUG("[kw2xrf] TC3TMOUT, SEQIRQ, TX failed\n");
             netdev->event_callback(netdev, NETDEV_EVENT_TX_NOACK);
-        } else {
-            DEBUG("[kw2xrf] SEQIRQ\n");
+        } else if(!channel_was_busy) {
```

Comment:
Why do you need to check here if the channel was busy before? Isn't this the case where an ACK was received?

Reply:
From my understanding there are two scenarios, where a TMR3IRQ does NOT occur:
Either the ACK came in time (success) OR the initial CCA failed and aborted the sequence (failure).

Reply:
Then please leave a small comment that explains the situation in the `else if`